File: python/examples/direct_payment.py
import os

from piaxis_sdk import PiaxisClient


piaxis = PiaxisClient(
    api_key=os.environ["PIAXIS_API_KEY"],
    base_url=os.getenv("PIAXIS_API_BASE_URL", "https://sandbox.api.gopiaxis.com/api"),
)

otp = piaxis.request_otp(
    {
        "email": "buyer@example.com",
        "phone_number": "+256700000000",
    }
)
print("OTP dispatch result:", otp)

payment = piaxis.create_payment(
    {
        "amount": "15000",
        "currency": "UGX",
        "payment_method": "mtn",
        "user_info": {
            "email": "buyer@example.com",
            "phone_number": "+256700000000",
            "otp": os.getenv("PIAXIS_TEST_OTP", "123456"),
        },
        "customer_pays_fees": True,
    }
)

print(payment)

File: examples/direct-payment.mjs
import { PiaxisClient } from "@piaxis/sdk";

const piaxis = new PiaxisClient({
  apiKey: process.env.PIAXIS_API_KEY,
  baseUrl: process.env.PIAXIS_API_BASE_URL ?? "https://sandbox.api.gopiaxis.com/api",
});

const otp = await piaxis.requestOtp({
  email: "buyer@example.com",
  phoneNumber: "+256700000000",
});

console.log("OTP dispatch result:", otp);

const payment = await piaxis.createPayment({
  amount: "15000",
  currency: "UGX",
  paymentMethod: "mtn",
  userInfo: {
    email: "buyer@example.com",
    phone_number: "+256700000000",
    otp: process.env.PIAXIS_TEST_OTP ?? "123456",
  },
  customerPaysFees: true,
});

console.log(payment);
